segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/stat/AutoScaleProcessorTest.java

```diff
@@ -367,4 +390,37 @@ protected void setTimeMillis(long time) {
         }
     }
 
+    @Test
+    public void testCacheGetOnReport() {
+        final int INFINITE_SIZE = Integer.MAX_VALUE;
+        final AtomicLong currentTime = new AtomicLong();
+        val evictions = new ArrayList<Map.Entry<Long, Long>>();
+        io.pravega.common.util.SimpleCacheTests.TestSimpleCache<String, Pair<Long, Long>> testSimpleCache =
+                new SimpleCacheTests.TestSimpleCache(
+                        INFINITE_SIZE,
+                        Duration.ofSeconds(60), (k, v) -> evictions.add(new AbstractMap.SimpleImmutableEntry(k, v)),
+                        currentTime::get);
+        TestAutoScaleProcessor monitor = new TestAutoScaleProcessor(
+                AutoScalerConfig.builder().with(AutoScalerConfig.MUTE_IN_SECONDS, 0)
+                        .with(AutoScalerConfig.COOLDOWN_IN_SECONDS, 0)
+                        .with(AutoScalerConfig.AUTH_ENABLED, authEnabled)
+                        .with(AutoScalerConfig.CACHE_CLEANUP_IN_SECONDS, 150)
+                        .with(AutoScalerConfig.CACHE_EXPIRY_IN_SECONDS, 60).build(),
+                executorService(), testSimpleCache);
+        String streamSegmentName1 = NameUtils.getQualifiedStreamSegmentName(SCOPE, STREAM1, 0L);
+        monitor.setTimeMillis(0L);
+        currentTime.set(0L);
+        monitor.notifyCreated(streamSegmentName1);
+        monitor.put(streamSegmentName1, new ImmutablePair<>(5L, 5L));
+        monitor.setTimeMillis(30L);
+        currentTime.set(30L);
+        monitor.report(streamSegmentName1, 10L, 0L, 10D, 10D, 10D, 10D);
+        monitor.setTimeMillis(80L);
+        currentTime.set(80L);
```

Comment:
set time such that T2 - T1 is > expiry time.. note expiry time that you have set in the test is 60000 ms. 
and here monitor.setTimeMillis is stting the time in millis.

Reply:
shouldn't T2-T0 > expiry time while T2-T1 < expiry time so that we can conclude that report updates latest access time successfully?

Reply:
sorry, yes, t2-t0 ? expiry.